pkg/container-utils/cri/cri_test.go

```diff
@@ -0,0 +1,233 @@
+// Copyright 2019-2022 The Inspektor Gadget authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package cri
+
+import (
+	"testing"
+
+	runtimeclient "github.com/kinvolk/inspektor-gadget/pkg/container-utils/runtime-client"
+)
+
+func TestParseExtraInfo(t *testing.T) {
+	type Mount struct {
+		src string
+		dst string
+	}
+	type Expected struct {
+		pid        *int
+		cgroupsPath *string
+		mounts     *[]Mount
+	}
```

Comment:
This allows you to *not* verify all fields.
If value is `nil` then value will not be verified.
If value is a valid pointer (as returned from `expect`) then it will be verified.

The reasoning is that not all tests have a valid pid/cgroupspath/mounts and you only want to verify them if the test has set them.

Reply:
Make sense. Thanks for the explanation.